cpp/modules/luma.gl/core/src/animation-loop-factory.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020 Unfolded Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#ifndef LUMAGL_CORE_ANIMATION_LOOP_FACTORY_H
+#define LUMAGL_CORE_ANIMATION_LOOP_FACTORY_H
+
+#include <memory>
+#include <optional>
+
+#include "luma.gl/core.h"
+
+namespace lumagl {
+
+struct AnimationLoopFactory {
```

Comment:
Another pattern is to have the `AnimationLoop` contain an `AnimationLoopImpl` or `AnimationLoop::Impl` and subclass that - that way we could avoid the factory pattern.

Reply:
In that case it would be `AnimationLoop` that decides which implementation to use based on the options that are passed?

I'm on the fence about options in general, if we're making the user specify them, we might as well have them provide an animation loop that `Deck` would use... The problem right now is that animation loops take different arguments based on their needs